src/postAggregation.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+var Promise = require('q')
+var _ = require('./lodash')
+
+var aggregation = require('./aggregation')
+
+module.exports = function(service) {
+  return {
+    post: post,
+    sortByKey: sortByKey,
+    limit: limit,
+    squash: squash,
+  }
+
+  function post(query, parent, cb) {
+    query.data = parent.locked ? _.clone(parent.data) : parent.data
```

Comment:
Feel like we should split this common line out. It's used everywhere and we always want it to be the same. Maybe a `getData` method?